File: test-api.py
import requests
import json

def test_matching_api():
    url = 'http://localhost:3002/api/matching'
    
    payload = {
        'apiKey': 'test-key',
        'players': [
            {
                'id': 'player1',
                'interests': ['RPG', 'Strategy'],
                'skillLevel': 7
            },
            {
                'id': 'player2',
                'interests': ['RPG', 'MOBA'],
                'skillLevel': 8
            }
        ],
        'groupSize': 2,
        'optimizationGoal': 'balanced'
    }
    
    headers = {'Content-Type': 'application/json'}
    
    try:
        print('Sending request to API...')
        response = requests.post(url, json=payload, headers=headers)
        
        print(f'Response status code: {response.status_code}')
        
        if response.status_code == 200:
            try:
                data = response.json()
                print('API Response:')
                print(json.dumps(data, indent=2))
            except json.JSONDecodeError:
                print('Error parsing JSON response')
                print('Raw response:', response.text)
        else:
            print('Error response:', response.text)
    
    except Exception as e:
        print(f'Error: {e}')

if __name__ == '__main__':
    test_matching_api() 

File: test-api.js
// Simple script to test the matching API endpoint
const fetch = require('node-fetch');

async function testMatchingAPI() {
  try {
    console.log('Sending request to API...');
    
    const requestBody = {
      apiKey: 'test-key',
      players: [
        {
          id: 'player1',
          interests: ['RPG', 'Strategy', 'FPS'],
          communicationStyle: 'casual',
          platformPreference: 'PC',
          playTimes: ['evening', 'weekend'],
          language: 'en',
          skillLevel: 7,
          contentTolerance: 8,
          themePreference: 'Action'
        },
        {
          id: 'player2',
          interests: ['RPG', 'MOBA', 'Strategy'],
          communicationStyle: 'competitive',
          platformPreference: 'PC',
          playTimes: ['evening', 'night'],
          language: 'en',
          skillLevel: 8,
          contentTolerance: 6,
          themePreference: 'Action'
        }
      ],
      groupSize: 2,
      optimizationGoal: 'balanced'
    };
    
    console.log('Request body:', JSON.stringify(requestBody, null, 2));
    
    const response = await fetch('http://localhost:3002/api/matching', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(requestBody)
    });
    
    console.log('Response status:', response.status);
    
    const text = await response.text();
    console.log('Raw response:', text);
    
    try {
      const data = JSON.parse(text);
      console.log('API Response (parsed):');
      console.log(JSON.stringify(data, null, 2));
    } catch (parseError) {
      console.error('Error parsing JSON response:', parseError);
    }
  } catch (error) {
    console.error('Error testing API:', error);
  }
}

testMatchingAPI(); 